sdk/core/azure-core/src/http/telemetry_policy.cpp

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <http/policy.hpp>
+
+#include <sstream>
+
+#ifdef _WIN32
+
+#define NOMINMAX
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+
+namespace {
+
+std::string GetOSVersion()
+{
+  std::ostringstream osVersionInfo;
+
+  {
+    HKEY regKey{};
+    auto regKeyOpened = false;
+    try
+    {
+      if (RegOpenKeyExA(
+              HKEY_LOCAL_MACHINE,
+              "SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion",
+              0,
+              KEY_READ,
+              &regKey)
+          == ERROR_SUCCESS)
+      {
+        regKeyOpened = true;
+
+        auto first = true;
+        char const* regValues[]{
+            "ProductName", "CurrentVersion", "CurrentBuildNumber", "BuildLabEx"};
+        for (auto regValue : regValues)
+        {
+          char valueBuf[200] = {};
+          DWORD valueBufSize = sizeof(valueBuf);
+
+          if (RegQueryValueExA(regKey, regValue, NULL, NULL, (LPBYTE)valueBuf, &valueBufSize)
+              == ERROR_SUCCESS)
+          {
+            if (valueBufSize > 0)
+            {
+              osVersionInfo << (first ? "" : " ")
+                            << std::string(valueBuf, valueBuf + (valueBufSize - 1));
+              first = false;
+            }
+          }
+        }
+      }
+
+      RegCloseKey(regKey);
+    }
+    catch (...)
+    {
+      if (regKeyOpened)
+      {
+        RegCloseKey(regKey);
+      }
+
+      throw;
+    }
+  }
+
+  return osVersionInfo.str();
+}
+
+} // namespace
+
+#else
+
+#include <sys/utsname.h>
+
+namespace {
```

Comment:
If GetOSVersion doesn't move to Utilities namespace then it needs to be in Azure::Core::Http::Details namespace

Reply:
it is in unnamed namespace. Detail namespace is only for headers.

Reply:
Because you can't (shouldn't) do unnamed namespaces in headers - they don't have the intended effect, and at the same time you need to make things available in the headers, but to your code. If the declaration is in the unnamed namespace in .cpp file, it is perfectly accessible the the code in the .cpp file, and only to that code, and the symbolas are not exported by linker (i.e. it is similar to `static` functions in C)